roles/vector/tasks/main.yml

```diff
@@ -37,7 +37,7 @@
 
 - name: Copy config
   template:
-    src: templates/vector.toml.j2
+    src: "{{ vector_template }}"
     dest: /etc/vector/vector.toml
```

Comment:
Maybe you want to change the destination too if you want to use yaml format

Reply:
"Tempus fugit" as they probably said in Rome... at some point...
Yes fair point, and one that's possibly a bit more complicated than first glance. I see two options...
1) Have the role user define the destination filename (or format) in an additional variable
2) Get ansible to strip the .j2 from the template filename and use whatever's left.
I've opted for 1. An extra variable, but its simpler and doesnt add any complexity to the unit tests.
see what you think. (commit is on its way)